### PR TITLE
reusable workflow to deploy Next.js to Pages 

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
In this PR, I have changed the `publish.yml` workflow in order to implement the reusable workflow `publish-next-on-ghpages.yml`.

### CODE REVIEW

**Review by AleCandido — CHANGES_REQUESTED** [review]
I forgot to submit this review some time ago

**Comment by AleCandido** [comment]
Apart from the very minor point above, is there anything else blocking this?

**Comment by AleCandido** [comment]
> Apart from the very minor point above, is there anything else blocking this?

@Edoardo-Pedicillo @MatteoRobbiati

**Comment by MatteoRobbiati** [comment]
> Apart from the very minor point above, is there anything else blocking this?

I think it can be merged.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,73 +1,12 @@
...
-    environment:
-      name: github-pages
-      url: ${{ steps.deployment.outputs.page_url }}
-    runs-on: ubuntu-latest
-    needs: build
-    steps:
-      - name: Deploy to GitHub Pages
-        id: deployment
-        uses: actions/deploy-pages@v1
+  deploy-docs:
```
[the remark this answers is not in the document]
Comment (AleCandido) [reply .github/workflows/publish.yml:11]:
```suggestion
  publish:
```

**Review by AleCandido — APPROVED** [review]

**Pull request merged**